x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/BlobCacheMetrics.java

```diff
@@ -7,15 +7,40 @@
 
 package org.elasticsearch.blobcache;
 
+import org.elasticsearch.index.shard.ShardId;
 import org.elasticsearch.telemetry.TelemetryProvider;
 import org.elasticsearch.telemetry.metric.LongCounter;
 import org.elasticsearch.telemetry.metric.LongHistogram;
 import org.elasticsearch.telemetry.metric.MeterRegistry;
 
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
 public class BlobCacheMetrics {
+    private static final String CACHE_POPULATION_REASON_ATTRIBUTE_KEY = "cachePopulationReason";
+    private static final String SHARD_ID_ATTRIBUTE_KEY = "shardId";
+
     private final LongCounter cacheMissCounter;
     private final LongCounter evictedCountNonZeroFrequency;
     private final LongHistogram cacheMissLoadTimes;
+    private final LongHistogram cachePopulateReadThroughput;
+    private final LongHistogram cachePopulateWriteThroughput;
+    private final LongHistogram cachePopulationElapsedTime;
+
+    public enum CachePopulationReason {
+        /**
+         * When warming the cache
+         */
+        Warming,
+        /**
+         * When fetching a new commit
+         */
+        LoadCommit,
+        /**
+         * When the data we need is not in the cache
+         */
+        CacheMiss
```

Comment:
`Warming` is good. But I am not sure about the other names. In theory, they are just non-warming triggered by all sorts of activities, such as opening engine, indexing, search etc. Just tossing some random idea, what about IndexInput?

Reply:
Yeah I will update these, I had `LoadCommit` in there for the `downloadCommit` use-case (which, as you pointed out, is disabled).

Am I right in thinking the other loads you mentioned are all triggered by a cache-miss? i.e. the caller asks the `SearchIndexInput` for the bytes and it loads any missing bytes from the blob-store? (that's why I called it `CacheMiss`)

What about `OnDemand`?, that would seem to cover anything that was loaded because it was needed immediately (i.e. not warming)

Reply:
In theory, it's always triggered by cache miss other than warming. We likely will load more than what is needed for that particular cache miss. So there will be data loaded but not needed immediately. But they are still triggered by cache miss.